com.unity.netcode.gameobjects/Tests/Runtime/NetworkTransform/NetworkTransformStateTests.cs

```diff
@@ -4,15 +4,87 @@
 
 namespace Unity.Netcode.RuntimeTests
 {
+
+    [TestFixture(TransformSpace.World)]
+    [TestFixture(TransformSpace.Local)]
     public class NetworkTransformStateTests
     {
+        public enum SyncAxis
+        {
+            SyncPosX,
+            SyncPosY,
+            SyncPosZ,
+            SyncPosXY,
+            SyncPosXZ,
+            SyncPosYZ,
+            SyncPosXYZ,
+            SyncRotX,
+            SyncRotY,
+            SyncRotZ,
+            SyncRotXY,
+            SyncRotXZ,
+            SyncRotYZ,
+            SyncRotXYZ,
+            SyncScaleX,
+            SyncScaleY,
+            SyncScaleZ,
+            SyncScaleXY,
+            SyncScaleXZ,
+            SyncScaleYZ,
+            SyncScaleXYZ,
+            SyncAllX,
+            SyncAllY,
+            SyncAllZ,
+            SyncAllXY,
+            SyncAllXZ,
+            SyncAllYZ,
+            SyncAllXYZ
+        }
+
+        public enum TransformSpace
+        {
+            World,
+            Local
+        }
+
+        public enum SynchronizationType
+        {
+            Delta,
+            Teleport
+        }
+
+        private TransformSpace m_TransformSpace;
+
+        public NetworkTransformStateTests(TransformSpace transformSpace)
+        {
+            m_TransformSpace = transformSpace;
+        }
+
+        private bool WillAnAxisBeSynchronized(ref NetworkTransform networkTransform)
+        {
+            return networkTransform.SyncScaleX || networkTransform.SyncScaleY || networkTransform.SyncScaleZ ||
+                networkTransform.SyncRotAngleX || networkTransform.SyncRotAngleY || networkTransform.SyncRotAngleZ ||
+                networkTransform.SyncPositionX || networkTransform.SyncPositionY || networkTransform.SyncPositionZ;
+        }
+
         [Test]
-        public void TestSyncAxes(
-            [Values] bool inLocalSpace,
-            [Values] bool syncPosX, [Values] bool syncPosY, [Values] bool syncPosZ,
-            [Values] bool syncRotX, [Values] bool syncRotY, [Values] bool syncRotZ,
-            [Values] bool syncScaX, [Values] bool syncScaY, [Values] bool syncScaZ)
+        public void TestSyncAxes([Values] SynchronizationType synchronizationType, [Values] SyncAxis syncAxis)
+
         {
+            bool inLocalSpace = m_TransformSpace == TransformSpace.Local;
+            bool isTeleporting = synchronizationType == SynchronizationType.Teleport;
```

Comment:
Don't hate me for the below code... :)
This was the best way I could come up with that would be sure to cover all of the possible combinations to assure each axial sync bool value was set when it needed to be set.
We could have more detailed Assert messages, but for now we know specifically what set of synchronized axis failed, whether it was when teleporting or just a delta update, and what space (world or local) it occurred within by the enum  parameters used.